### ISSUE TITLE
RefineNet Implementation

### ISSUE THREAD

**Issue by Harry-Zhi** [GeorgeSeif/Semantic-Segmentation-Suite#39]
Hi GeorgeSeif,

Thanks for your implementation of so many SOTA segmentation networks.

I am interested in the RefineNet which works quite well on many datasets.

(1)I notice that the original paper of RefineNet states that we should  first apply the softmax function to the 1/4 scale output of RefineNet blocks to get a dense score map, and then upscale the score map to the original size.

In your implementation, you first upscale and then calculate softmax cross entropy loss.
Have you tested the effect of these two implementations?

(2)Similarly, the author applied another RCU model to the final output of RefineNet blocks before upscaling and softmax. However, your implementation seems to miss these two RCU units.

(3)Did you try to obtain some performance of RefineNet on popular datasets such as NYUv2, CamVid or CityScape, compared to the original paper?

If so, would you mind telling me the performance of your implementation? How about the efficiency (inference time on a modern GPU using ResNet50/101?)

Many thanks.

Shuaifeng

**Comment by GeorgeSeif** [comment]
Hi there!

(1) I actually did notice this a few weeks ago for a number of works that output at a lower resolution and how my code does that differently as you stated. I decided not to change the code since when I did run it the results were not different at all. You will get a tiny speed boost from doing the way the paper does though, since you're final computations are applied at low res, but the speed boost is tiny because you're only doing a 1x1 softmax. You can switch it around to the paper's method if you like.

(2) Ah you are very correct there! I'll update it now! That's what I get when I read too fast!

(3) I tested on CamVid but I don't remember the exact final results. I usually don't train it fully with fine tuning, data augmentation etc. I just train with regular dataset for 100 epochs or so, which will get some good results to show me that the implementation is working well. Plus by that point I can still see the difference in performance between methods, without actually training them for super long and with tons of data,

Please let me know if you have any further questions on it!

**Comment by Harry-Zhi** [comment]
Thanks for your explanation, GeorgeSeif

**Comment by nooriahmed** [comment]
How to train on cityscape dataset? instead of CamVid.. --model MobileUNet --frontend ResNet101. My question might be silly and basic but it has a lot of meanings for me. should I just give path and argument to the said dataset? any response would be highly appreciated.  Regards
